packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -64,15 +94,103 @@ export default class Sayt extends LitElement {
     this.visible = true;
   }
 
+  /**
+   * Makes SAYT visible if the event refers to the correct SAYT component.
+   *
+   * @param event An event that can contain a searchbar ID.
+   */
+  showCorrectSayt(event: CustomEvent) {
+    if (this.isCorrectSayt(event)) {
+      this.showSayt();
+    }
+  }
+
   /**
    * Changes the `visible` property to `false`.
    */
   hideSayt() {
     this.visible = false;
   }
 
+  /**
+   * Hides SAYT if the event refers to the correct SAYT component.
+   *
+   * @param event An event that can contain a searchbar ID.
+   */
+  hideCorrectSayt(event: CustomEvent) {
+    if (this.isCorrectSayt(event)) {
+      this.hideSayt();
+    }
+  }
+
+  /**
+   * Determines whether an event refers to the correct SAYT. This is true if
+   * a matching `searchbar` ID is specified, if the event has no `searchbar`
+   * ID specified, or if SAYT has no `searchbar` ID specified.
+   *
+   * @param event An event that can contain a searchbar ID for comparison.
+   */
+  isCorrectSayt(event: CustomEvent) {
+    const searchbar = event.detail && event.detail.searchbar;
+    return this.searchbar === undefined ||
+      searchbar === this.searchbar ||
+      searchbar === undefined;
+  }
+
+  /**
+   * Processes a click event in order to close SAYT under the right conditions.
+   * 
+   * @param event The click event.
+   */
+  processClick(event: Event) {
+    const target = event.target as Node;
+    if (this.contains(target) || this.nodeInSearchBar(target)) return;
+
+    this.hideSayt();
+  }
+
+  /**
+   * Handles hiding SAYT on click on a close link/button (or other event).
+   *
+   * @param event An event with a default action to be prevented.
+   */
+  clickCloseSayt(event: Event) {
+    event.preventDefault();
+    this.hideSayt();
+  }
+
+  /**
+   * Checks whether a given node is inside of SAYT's identified search bar.
+   * 
+   * @param node The node to check for containment.
+   */
+  nodeInSearchBar(node: Node) {
+    if (!this.searchbar) return false;
+    const searchBar = document.querySelector('#' + this.searchbar);
+    return !!searchBar && searchBar.contains(node);
+  }
+
+  /**
+   * Processes a keypress event in order to close SAYT under the right conditions.
+   * @param event 
+   */
+  processKeyPress(event: KeyboardEvent) {
+    if (event.key === "Escape") {
+      this.hideSayt();
+    }
+  }
+
+  /**
+   * Returns a TemplateResult object for rendering in LitElement.
+   */
   render() {
     return html`
+      ${ this.showCloseButton ?
+        html`<a href="#" aria-label="Close" .onclick=${ this.clickCloseSayt }>
```

Comment:
Lit-Element has @event bindings. We're currently using @click, @input, & @keydown in the autocomplete and search-box components.